tests_integ/helpers/deployer/utils/time.py

```diff
@@ -0,0 +1,127 @@
+"""
+Date & Time related utilities
+"""
+
+import datetime
+import dateparser
+
+from dateutil.tz import tzutc
+
+
+def timestamp_to_iso(timestamp):
+    """
+    Convert Unix Epoch Timestamp to ISO formatted time string:
+        Ex: 1234567890 -> 2018-07-05T03:09:43.842000
+
+    Parameters
+    ----------
+    timestamp : int
+        Unix epoch timestamp
+
+    Returns
+    -------
+    str
+        ISO formatted time string
+    """
+
+    return to_datetime(timestamp).isoformat()
+
+
+def to_datetime(timestamp):
+    """
+    Convert Unix Epoch Timestamp to Python's ``datetime.datetime`` object
+
+    Parameters
+    ----------
+    timestamp : int
+        Unix epoch timestamp
+
+    Returns
+    -------
+    datetime.datetime
+        Datetime representation of timestamp
+    """
+
+    timestamp_secs = int(timestamp) / 1000.0
+    return datetime.datetime.utcfromtimestamp(timestamp_secs)
+
+
+def to_timestamp(some_time):
+    """
+    Converts the given datetime value to Unix timestamp
+
+    Parameters
+    ----------
+    some_time : datetime.datetime
+        Value to be converted to unix epoch. This must be without any timezone identifier
+
+    Returns
+    -------
+    int
+        Unix timestamp of the given time
+    """
+
+    # `total_seconds()` returns elaped microseconds as a float. Get just milliseconds and discard the rest.
+    return int((some_time - datetime.datetime(1970, 1, 1)).total_seconds() * 1000.0)
```

Comment:
Why not [`datetime.timestamp`](https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp)?

Reply:
It doesn't produce the right value:
```python
>>> d = datetime.datetime(2020,12,18)
>>> d
datetime.datetime(2020, 12, 18, 0, 0)
>>> d.timestamp() * 1000
1608278400000.0 # 2020-12-18 at 8 AM
# Current code:
>>> int((d - datetime.datetime(1970, 1, 1)).total_seconds() * 1000.0)
1608249600000 # 2020-12-18 at 12 AM
# From the docs https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp:
>>> ((d - datetime.datetime(1970, 1, 1)) / datetime.timedelta(seconds=1)) * 1000
1608249600000.0 # 2020-12-18 at 12 AM
```

Reply:
Ah fair enough, and that's mentioned in the docs.